### PR TITLE
Initial implementation of tpu_config in container cluster

### PR DESCRIPTION
**Pull request by c2thorn**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/11748

The API does not return `tpu_config` a this time, but moving forward without reading. Added https://github.com/hashicorp/terraform-provider-google/issues/11880 in the interim

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: added `tpu_config` to `google_container_cluster` (beta only)
```

### CODE REVIEW

File: mmv1/third_party/terraform/resources/resource_container_cluster.go.erb [hunk 1]
```diff
@@ -543,10 +543,47 @@ func resourceContainerCluster() *schema.Resource {
 				ForceNew:    true,
 				Description: `Whether to enable Cloud TPU resources in this cluster.`,
 <% unless version == 'ga' -%>
-				Default: false,
+				ExactlyOneOf: []string{
+					"enable_tpu",
+					"tpu_config",
+				},
+				Computed: true,
```
Comment (melinath) [inline mmv1/third_party/terraform/resources/resource_container_cluster.go.erb:550]: IIRC we consider this non-breaking as long as the API is currently returning false as the default?

**Review by melinath — CHANGES_REQUESTED** [review]
LGTM apart from the issue with the tests.

File: mmv1/third_party/terraform/resources/resource_container_cluster.go.erb (hunk 1 above)
Comment (c2thorn) [reply mmv1/third_party/terraform/resources/resource_container_cluster.go.erb:547]: yes that's what I understand. The API doesn't return this field at all when not specified either with this field or `tpu_config`

File: mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb [hunk 2]
```diff
@@ -5321,3 +5347,62 @@ resource "google_container_cluster" "primary" {
...
+		ip_cidr_range = "10.0.0.0/19"
+	}
+
+	secondary_ip_range {
+		range_name    = "svc"
+		ip_cidr_range = "10.0.32.0/22"
+	}
+}
+	
+resource "google_container_cluster" "with_tpu" {
```
Comment (shuyama1) [inline mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb:5378]:
```suggestion
resource "google_container_cluster" "with_tpu_config" {
```

**Review by shuyama1 — CHANGES_REQUESTED** [review]

**Review by shuyama1 — APPROVED** [review]
LGTM assuming tests pass.

**Pull request merged**